stdlib/result.ml

```diff
@@ -0,0 +1,43 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+type ('a, 'e) t = ('a, 'e) result = Ok of 'a | Error of 'e
+
+let ok v = Ok v
+let error e = Error e
+let value r ~default = match r with Ok v -> v | Error _ -> default
+let get_ok = function Ok v -> v | Error _ -> invalid_arg "result is Error _"
+let get_error = function Error e -> e | Ok _ -> invalid_arg "result is Ok _"
+let bind r f = match r with Ok v -> f v | Error _ as e -> e
+let join = function Ok r -> r | Error _ as e -> e
+let map f = function Ok v -> Ok (f v) | Error _ as e -> e
+let fold ok error = function Ok v -> ok v | Error e -> error e
+let iter f = function Ok v -> f v | Error _ -> ()
+let recover f r = function Ok _ as v -> v | Error e -> Ok (f v)
```

Comment:
Shouldn't that read `| Error e -> Ok (f e)` ??

Reply:
Oh, and based on the docstring, shouldn't it read `Ok _ as v -> r`? Or is the docstring wrong?

Reply:
No, pardon, I had that last bit wrong. Still, the function is wrong, `r` isn't being used correctly I think?

Reply:
Not crazy. Forgot to compile before pushing... Thanks.